amf-webapi/shared/src/main/scala/amf/plugins/document/webapi/parser/spec/domain/ServerEmitters.scala

```diff
@@ -261,9 +261,14 @@ private case class Servers(default: Option[Server], servers: Seq[Server])
 
 private object Servers {
   def apply(f: FieldEntry): Servers = {
-    val (default, servers) =
-      f.arrayValues(classOf[Server]).partition(_.annotations.find(classOf[SynthesizedField]).isDefined)
-    new Servers(default.headOption, servers)
+    f.arrayValues(classOf[Server]).toList match {
```

Comment:
Simplify method as complexity has definitely increased

Reply:
We need to check if there is only one server before make the filter. We could check the length, and them do the filter, or just use the unapply logic of the List class.